crates/sats/src/db/def.rs

```diff
@@ -343,24 +332,118 @@ impl From<&ColumnSchema> for ProductTypeElement {
 #[derive(Clone)]
 pub struct ColumnDefMeta {
     pub column: ProductTypeElement,
-    pub attr: ColumnIndexAttribute,
+    pub attr: Constraints,
     pub pos: usize,
 }
 
-impl From<&ColumnSchema> for ColumnDefMeta {
-    fn from(value: &ColumnSchema) -> Self {
+/// Describe the columns + meta attributes
+/// TODO(cloutiertyler): This type should be deprecated and replaced with
+/// ColumnDef or ColumnSchema where appropriate
+#[derive(Debug, Clone, Eq, PartialEq, PartialOrd, Ord)]
+pub struct ProductTypeMeta {
+    pub columns: ProductType,
+    pub attr: Vec<Constraints>,
```

Comment:
Nit: should this field be named `constraints` now?

Reply:
This will be gone it the last PR, and for now `attr` is more correct because is for the column, not the table.